File: onyx-database-tests/src/main/java/entities/AbstractEntity.java
package entities;

import com.onyx.persistence.ManagedEntity;
import com.onyx.persistence.annotations.Attribute;

import java.util.Date;

/**
 * Created by timothy.osborn on 10/20/14.
 */
public abstract class AbstractEntity extends ManagedEntity
{

    @Attribute
    public Date dateCreated;

    @Attribute
    public Date dateUpdated;

    public double doubleSample;
    public Double dblSmaple;

}

File: onyx-database-tests/src/main/kotlin/entities/AbstractEntity.kt
package entities

import com.onyx.persistence.ManagedEntity
import com.onyx.persistence.annotations.Attribute

import java.util.Date

/**
 * Created by timothy.osborn on 10/20/14.
 */
@Suppress("unused")
abstract class AbstractEntity : ManagedEntity() {

    @Attribute
    var dateCreated: Date? = null

    @Attribute
    var dateUpdated: Date? = null

    var doubleSample: Double = 0.toDouble()
    var dblSample: Double? = null

}
